File: venda/src/core/oop/produto-digital.ts
import { ItemVenda } from "./item-venda.js";

export class ProdutoDigital extends ItemVenda {
  constructor(
    id: string,
    nome: string,
    preco: number,
    public tamanhoMB: number
  ) {
    super(id, nome, preco);
  }

  getTipo(): string {
    return "Produto Digital";
  }
}

File: venda/projeto_vendas_1/servidor-java/ProdutoDigital.java

public class ProdutoDigital extends ItemVenda {
    private String linkDownload;

    public ProdutoDigital(int id, String nome, double preco, String linkDownload) {
        super(id, nome, preco);
        this.linkDownload = linkDownload;
    }

    public String getLinkDownload() {
        return linkDownload;
    }
}